backend/src/data-source.ts

```diff
@@ -343,45 +351,65 @@ export class Database {
 	public async lookupBacklogById(id: number): Promise<BacklogItem> {
 		const maybeBacklog =  await this.dataSource.manager.findOneBy(BacklogItem, {id: id});
 		if (!maybeBacklog) {
-			throw new Error(`TODO with id ${id} not found`)
+			throw new NotFoundError(`TODO with id ${id} not found`)
 		}
 		return maybeBacklog
 	}
 	
 	public async lookupStoryById(id: number): Promise<Story> {
 		const maybeStory =  await this.dataSource.manager.findOneBy(Story, {id: id});
 		if (!maybeStory) {
-			throw new Error(`TODO with id ${id} not found`)
+			throw new NotFoundError(`TODO with id ${id} not found`)
 		}
 		return maybeStory
 	}
 
 	///// General Methods - Only use if there is not a method above to use /////
 
 	public async save(item: any) {
-		return await this.dataSource.manager.save(item);
+		try {
+			return await this.dataSource.manager.save(item);
+		} catch {
+			throw new NotSavedError(`Failed to save ${item}`)
+		}
 	}
```

Comment:
The rest of the class calls this save method so we don't have to check every time any method calls a save